Classes/common/CDTDatastore.m

```diff
@@ -396,6 +396,25 @@ - (BOOL)validateBodyDictionary:(NSDictionary *)body error:(NSError *__autoreleas
             return NO;
         }
     }
+    
+    //now to check if the body contains valid json data
+    if(![NSJSONSerialization isValidJSONObject:body]){
+        //body isn't valid json, set error
+        if (error){
+            NSInteger code = 400;
```

Comment:
Shouldn't you use `kTDStatusBadRequest` if you must use 400, or perhaps `kTDStatusBadJSON` would be even better